src/components/internal/DatePicker/DatePickerOverlay.tsx

```diff
@@ -1,26 +1,49 @@
 import React, { HTMLAttributes } from "react";
 import { OverlayTriggerState } from "react-stately";
 import { DatePicker, DatePickerProps } from "src/components/internal/DatePicker/DatePicker";
+import { DateRangePicker, DateRangePickerProps } from "src/components/internal/DatePicker/DateRangePicker";
 import { Css } from "src/Css";
+import { DateFieldMode } from "src/inputs";
 
-interface DatePickerOverlayProps extends DatePickerProps {
+interface BaseDatePickerOverlayProps {
   overlayProps: HTMLAttributes<HTMLElement>;
   state: OverlayTriggerState;
+  mode?: DateFieldMode;
+}
+
+// interface DatePickerOverlayProps extends Omit<DatePickerProps, "value">, BaseDatePickerOverlayProps {
+interface DatePickerOverlayProps extends DatePickerProps, BaseDatePickerOverlayProps {
+  mode: "single";
+}
+// interface DateRangePickerOverlayProps extends Omit<DateRangePickerProps, "range">, BaseDatePickerOverlayProps {
+interface DateRangePickerOverlayProps extends DateRangePickerProps, BaseDatePickerOverlayProps {
+  mode: "range";
 }
 
 // Small wrapper around DatePicker to provide necessary styling and state handling when displayed as an overlay.
-export function DatePickerOverlay({ overlayProps, state, ...datePickerProps }: DatePickerOverlayProps) {
+export function DatePickerOverlay(props: DatePickerOverlayProps | DateRangePickerOverlayProps) {
```

Comment:
Thought: I wonder if `DatePickerOverlay` needs to really care about "am I wrapping a date picker or a date range picker", or if it could just take `childeren`?

Reply:
Yea, good thought. I like that idea.